controllers/propagator/propagation.go

```diff
@@ -945,6 +993,28 @@ func (r *PolicyReconciler) processTemplates(
 		}
 
 		policyT.ObjectDefinition.Raw = resolveddata
+
+		// Set initialization vector annotation on the ObjectDefinition for the controller's use
+		if usesEncryption {
+			policyTObjectUnstructured := &unstructured.Unstructured{}
+
+			jsonErr := json.Unmarshal(resolveddata, policyTObjectUnstructured)
+			if jsonErr != nil {
+				return fmt.Errorf("failed to unmarshal the object definition to JSON: %w", jsonErr)
+			} else {
```

Comment:
Could you please remove the else since your always return inside the if block? This makes it easier to read.